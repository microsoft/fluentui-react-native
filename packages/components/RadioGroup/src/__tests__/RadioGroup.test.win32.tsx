import * as React from 'react';
import { RadioGroup, RadioButton } from '..';
import * as renderer from 'react-test-renderer';

function onChange(key: string) {
  console.log(key);
}

describe('RadioButton Tests', () => {
  it('RadioButton default', () => {
    const tree = renderer.create(<RadioButton content="Test" buttonKey="a" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Disabled RadioButton', () => {
    const tree = renderer.create(<RadioButton content="Test" buttonKey="b" disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('RadioButton with all props', () => {
    const tree = renderer.create(<RadioButton content="Test" buttonKey="a" disabled={false} ariaLabel="Test ariaLabel" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('RadioButton with all props and tokens', () => {
    const TokenizedRadioButton = RadioButton.customize({
      tokens: { borderColor: 'blue', backgroundColor: 'black', color: 'red', fontSize: 18 }
    });
    const tree = renderer
      .create(<TokenizedRadioButton content="Test" buttonKey="a" disabled={false} ariaLabel="Test ariaLabel" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('RadioGroup + RadioButton Tests', () => {
  it('RadioGroup default', () => {
    const tree = renderer.create(<RadioGroup label="Test RadioGroup" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tokenized RadioGroup default', () => {
    const TokenizedRadioGroup = RadioGroup.customize({ tokens: { fontSize: 18, color: 'blue' } });
    const tree = renderer.create(<TokenizedRadioGroup label="Test RadioGroup" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('RadioGroup with all props and default RadioButton', () => {
    const tree = renderer
      .create(
        <RadioGroup label="Test RadioGroup" defaultSelectedKey="a" ariaLabel="RadioGroup ariaLabel" onChange={onChange}>
          <RadioButton content="Test" buttonKey="a" />
        </RadioGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('RadioGroup with all props and four default RadioButtons', () => {
    const tree = renderer
      .create(
        <RadioGroup label="This is a test RadioGroup" defaultSelectedKey="A" onChange={onChange}>
          <RadioButton content="Option A" buttonKey="A" ariaLabel="Cool" />
          <RadioButton content="Option B" buttonKey="B" />
          <RadioButton content="Option C" buttonKey="C" disabled={true} />
          <RadioButton content="Option D" buttonKey="D" />
        </RadioGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
