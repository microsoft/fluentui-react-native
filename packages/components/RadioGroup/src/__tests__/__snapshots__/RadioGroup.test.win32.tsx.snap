// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Tests Disabled RadioButton 1`] = `
<View
  acceptsKeyboardFocus={true}
  accessibilityActions={
    Array [
      Object {
        "label": "Select a RadioButton",
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Test"
  accessibilityRole="radio"
  accessibilityStates={
    Array [
      "disabled",
    ]
  }
  accessible={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  rest={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 0,
      "minHeight": 20,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#f3f2f1",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 20,
        "left": 0,
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 4,
        "top": 0,
        "width": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "height": 10,
          "left": 4,
          "opacity": 0,
          "position": "relative",
          "top": 4,
          "width": 10,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "borderColor": "transparent",
        "borderStyle": "dashed",
        "borderWidth": 1,
        "color": "#a19f9d",
        "fontSize": 12,
        "marginTop": 3,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`RadioButton Tests RadioButton default 1`] = `
<View
  acceptsKeyboardFocus={true}
  accessibilityActions={
    Array [
      Object {
        "label": "Select a RadioButton",
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Test"
  accessibilityRole="radio"
  accessibilityStates={Array []}
  accessible={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  rest={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 0,
      "minHeight": 20,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#323130",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 20,
        "left": 0,
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 4,
        "top": 0,
        "width": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#323130",
          "borderRadius": 5,
          "height": 10,
          "left": 4,
          "opacity": 0,
          "position": "relative",
          "top": 4,
          "width": 10,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "borderColor": "transparent",
        "borderStyle": "dashed",
        "borderWidth": 1,
        "color": "#323130",
        "fontSize": 12,
        "marginTop": 3,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`RadioButton Tests RadioButton with all props 1`] = `
<View
  acceptsKeyboardFocus={true}
  accessibilityActions={
    Array [
      Object {
        "label": "Select a RadioButton",
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Test ariaLabel"
  accessibilityRole="radio"
  accessibilityStates={Array []}
  accessible={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  rest={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 0,
      "minHeight": 20,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#323130",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 20,
        "left": 0,
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 4,
        "top": 0,
        "width": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#323130",
          "borderRadius": 5,
          "height": 10,
          "left": 4,
          "opacity": 0,
          "position": "relative",
          "top": 4,
          "width": 10,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "borderColor": "transparent",
        "borderStyle": "dashed",
        "borderWidth": 1,
        "color": "#323130",
        "fontSize": 12,
        "marginTop": 3,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`RadioButton Tests RadioButton with all props and tokens 1`] = `
<View
  acceptsKeyboardFocus={true}
  accessibilityActions={
    Array [
      Object {
        "label": "Select a RadioButton",
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Test ariaLabel"
  accessibilityRole="radio"
  accessibilityStates={Array []}
  accessible={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  rest={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 0,
      "minHeight": 20,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "blue",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 20,
        "left": 0,
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 4,
        "top": 0,
        "width": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 10,
          "left": 4,
          "opacity": 0,
          "position": "relative",
          "top": 4,
          "width": 10,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "borderColor": "transparent",
        "borderStyle": "dashed",
        "borderWidth": 1,
        "color": "red",
        "fontSize": 18,
        "marginTop": 3,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`RadioGroup + RadioButton Tests RadioGroup default 1`] = `
<View
  accessibilityLabel="Test RadioGroup"
  accessibilityRole="radiogroup"
  accessible={true}
  rest={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "inherit",
        "fontSize": 12,
        "fontWeight": "600",
      }
    }
  >
    Test RadioGroup
  </Text>
  <View />
</View>
`;

exports[`RadioGroup + RadioButton Tests RadioGroup with all props and default RadioButton 1`] = `
<View
  accessibilityLabel="RadioGroup ariaLabel"
  accessibilityRole="radiogroup"
  accessible={true}
  rest={
    Object {
      "defaultSelectedKey": "a",
      "onChange": [Function],
    }
  }
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "inherit",
        "fontSize": 12,
        "fontWeight": "600",
      }
    }
  >
    Test RadioGroup
  </Text>
  <View>
    <View
      acceptsKeyboardFocus={true}
      accessibilityActions={
        Array [
          Object {
            "label": "Select a RadioButton",
            "name": "Select",
          },
        ]
      }
      accessibilityLabel="Test"
      accessibilityRole="radio"
      accessibilityStates={
        Array [
          "selected",
        ]
      }
      accessible={true}
      onAccessibilityAction={[Function]}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      rest={Object {}}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 0,
          "minHeight": 20,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#323130",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 20,
            "left": 0,
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 4,
            "top": 0,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#323130",
              "borderRadius": 5,
              "height": 10,
              "left": 4,
              "opacity": 1,
              "position": "relative",
              "top": 4,
              "width": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "borderColor": "rgba(128, 128, 128, 1)",
            "borderStyle": "dashed",
            "borderWidth": 1,
            "color": "#323130",
            "fontSize": 12,
            "marginTop": 3,
          }
        }
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;

exports[`RadioGroup + RadioButton Tests RadioGroup with all props and four default RadioButtons 1`] = `
<View
  accessibilityLabel="This is a test RadioGroup"
  accessibilityRole="radiogroup"
  accessible={true}
  rest={
    Object {
      "defaultSelectedKey": "A",
      "onChange": [Function],
    }
  }
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "inherit",
        "fontSize": 12,
        "fontWeight": "600",
      }
    }
  >
    This is a test RadioGroup
  </Text>
  <View>
    <View
      acceptsKeyboardFocus={true}
      accessibilityActions={
        Array [
          Object {
            "label": "Select a RadioButton",
            "name": "Select",
          },
        ]
      }
      accessibilityLabel="Cool"
      accessibilityRole="radio"
      accessibilityStates={
        Array [
          "selected",
        ]
      }
      accessible={true}
      onAccessibilityAction={[Function]}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      rest={Object {}}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 0,
          "minHeight": 20,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#323130",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 20,
            "left": 0,
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 4,
            "top": 0,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#323130",
              "borderRadius": 5,
              "height": 10,
              "left": 4,
              "opacity": 1,
              "position": "relative",
              "top": 4,
              "width": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "borderColor": "rgba(128, 128, 128, 1)",
            "borderStyle": "dashed",
            "borderWidth": 1,
            "color": "#323130",
            "fontSize": 12,
            "marginTop": 3,
          }
        }
      >
        Option A
      </Text>
    </View>
    <View
      acceptsKeyboardFocus={true}
      accessibilityActions={
        Array [
          Object {
            "label": "Select a RadioButton",
            "name": "Select",
          },
        ]
      }
      accessibilityLabel="Option B"
      accessibilityRole="radio"
      accessibilityStates={Array []}
      accessible={true}
      onAccessibilityAction={[Function]}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      rest={Object {}}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 0,
          "minHeight": 20,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#323130",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 20,
            "left": 0,
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 4,
            "top": 0,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#323130",
              "borderRadius": 5,
              "height": 10,
              "left": 4,
              "opacity": 0,
              "position": "relative",
              "top": 4,
              "width": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "borderColor": "transparent",
            "borderStyle": "dashed",
            "borderWidth": 1,
            "color": "#323130",
            "fontSize": 12,
            "marginTop": 3,
          }
        }
      >
        Option B
      </Text>
    </View>
    <View
      acceptsKeyboardFocus={true}
      accessibilityActions={
        Array [
          Object {
            "label": "Select a RadioButton",
            "name": "Select",
          },
        ]
      }
      accessibilityLabel="Option C"
      accessibilityRole="radio"
      accessibilityStates={
        Array [
          "disabled",
        ]
      }
      accessible={true}
      onAccessibilityAction={[Function]}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      rest={Object {}}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 0,
          "minHeight": 20,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f3f2f1",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 20,
            "left": 0,
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 4,
            "top": 0,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "height": 10,
              "left": 4,
              "opacity": 0,
              "position": "relative",
              "top": 4,
              "width": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "borderColor": "transparent",
            "borderStyle": "dashed",
            "borderWidth": 1,
            "color": "#a19f9d",
            "fontSize": 12,
            "marginTop": 3,
          }
        }
      >
        Option C
      </Text>
    </View>
    <View
      acceptsKeyboardFocus={true}
      accessibilityActions={
        Array [
          Object {
            "label": "Select a RadioButton",
            "name": "Select",
          },
        ]
      }
      accessibilityLabel="Option D"
      accessibilityRole="radio"
      accessibilityStates={Array []}
      accessible={true}
      onAccessibilityAction={[Function]}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      rest={Object {}}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 0,
          "minHeight": 20,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#323130",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 20,
            "left": 0,
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 4,
            "top": 0,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#323130",
              "borderRadius": 5,
              "height": 10,
              "left": 4,
              "opacity": 0,
              "position": "relative",
              "top": 4,
              "width": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "borderColor": "transparent",
            "borderStyle": "dashed",
            "borderWidth": 1,
            "color": "#323130",
            "fontSize": 12,
            "marginTop": 3,
          }
        }
      >
        Option D
      </Text>
    </View>
  </View>
</View>
`;

exports[`RadioGroup + RadioButton Tests Tokenized RadioGroup default 1`] = `
<View
  accessibilityLabel="Test RadioGroup"
  accessibilityRole="radiogroup"
  accessible={true}
  rest={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "blue",
        "fontFamily": "inherit",
        "fontSize": 18,
        "fontWeight": "600",
      }
    }
  >
    Test RadioGroup
  </Text>
  <View />
</View>
`;
